src/PaperG/FirehoundBlob/Facebook/FacebookCreative.php

```diff
@@ -105,18 +126,26 @@ public function toArray()
      */
     public function fromArray($array)
     {
-        $this->type = $this->safeGet($array, self::TYPE);
-        $objects = $this->safeGet($array, self::OBJECTS, []);
-        $objectArray = [];
-        foreach ($objects as $object) {
-            if ($this->type == 'carousel') {
-                $objectArray[] = new FacebookCarouselCreativeData($object);
-            } else {
-                $objectArray[] = new FacebookCreativeData($object); //versioned, might need builder
-            }
+        $this->type       = $this->safeGet($array, self::TYPE);
+        $primary    = $this->safeGet($array, self::PRIMARY);
+        $childAttachments = $this->safeGet($array, self::CHILD_ATTACHMENTS, []);
+
+        $this->primary = $this->createCreativeData($primary);
+
+        $objectsArray = [];
+        foreach ($childAttachments as $object) {
+            $objectsArray[]  = $this->createCreativeData($object);
         }
 
-        $this->objects = $objectArray;
+        $this->childAttachments = $objectsArray;
     }
 
+    private function createCreativeData($creative)
+    {
+        if ($this->type == self::CAROUSEL_TYPE) {
+            return new FacebookCarouselCreativeData($creative);
+        } else {
+            return new FacebookCreativeData($creative);
+        }
+    }
```

Comment:
So... say we switch it up - `FacebookCreative` becomes an abstract class, extended by `FacebookCarouselCreative` and `FacebookLinkCreative`.  Now, I have a class that should contain a `FacebookCreative` as one of it's properties - how do I know which type of `FacebookCreative` it contains?  Do I then make a `FacebookCarouselCreativeContainer` and a `FacebookLinkCreativeContainer`?  Duplication cannot be the answer here.